reconcile/ocm_clusters.py

```diff
@@ -24,6 +24,42 @@ def fetch_desired_state(clusters):
     return desired_state
 
 
+def get_cluster_update_spec(cluster_name, current_spec, desired_spec):
+    """ Get a cluster spec to update. Returns an error if diff is invalid """
+    allowed_spec_update_fields = [
+        'instance_type',
+        'storage',
+        'load_balancers',
+        'private',
+        'channel',
+        'autoscale',
+        'nodes'
+    ]
+    error = False
+    if current_spec['network'] != desired_spec['network']:
+        error = True
+        logging.error(f'[{cluster_name}] invalid update: network')
+    current_spec_spec = current_spec['spec']
+    desired_spec_spec = desired_spec['spec']
+    updated = {k: desired_spec_spec[k] for k in desired_spec_spec
+               if current_spec_spec[k] != desired_spec_spec[k]}
+
+    # we only need deleted to check if a field removal is valid
+    # we really want to check updated + deleted, and since
+    # we have no further use for deleted -
+    deleted = {k: current_spec_spec[k] for k in current_spec_spec
+               if k not in desired_spec_spec}
+    diffs = deleted
+    diffs.update(updated)
+
+    for k in diffs:
```

Comment:
Easier if `allowed_spec_update_fields` is a set:

```python
invalid_fields = set(diffs.keys()) & allowed_spec_update_fields
error = True
logging.error(...)
```

Reply:
"easier" is subjective :)

Reply:
Sets are easy. Unless you wonder whether the set of all sets that contain themselves is a member of the set of all sets that contain themselves. ;)

Reply:
Russel's set?